axiom/tests/test_txn.c

```diff
@@ -8068,14 +8084,165 @@ static void test_segment_record_error(void) {
+                         nr_log_events_number_seen(txn->log_events));
+  tlib_pass_if_int_equal("null log level, event saved", 1,
+                         nr_log_events_number_saved(txn->log_events));
+  log_event_json = nr_log_events_get_event_json(txn->log_events, 0);
+  tlib_pass_if_not_null("null log level, event recorded", log_event_json);
+  expected
+      = "[{"
+        "\"message\":\"" LOG_MESSAGE
+        "\","
+        "\"log.level\":\"UNKNOWN\","
+        "\"timestamp\":0,"
+        "\"trace.id\":\"0000000000000000\","
+        "\"span.id\":\"0000000000000000\","
+        "\"entity.guid\":\"null\","
+        "\"entity.name\":\"" APP_ENTITY_NAME
+        "\","
+        "\"hostname\":\"null\""
+        "}]";
+  tlib_pass_if_str_equal("null log level, event recorded, json ok", expected,
+                         log_event_json);
+  test_txn_metric_is("null log level, event recorded, metric created",
+                     txn->unscoped_metrics, MET_FORCED, "Logging/lines", 1, 0,
+                     0, 0, 0, 0);
+  test_txn_metric_is("null log level, event recorded, metric created",
+                     txn->unscoped_metrics, MET_FORCED, "Logging/lines/UNKNOWN",
+                     1, 0, 0, 0, 0, 0);
+  nr_txn_destroy(&txn);
+
+  /* Happy path - everything initialized: record! */
+  txn = new_txn_for_record_log_event_test(APP_ENTITY_NAME);
+  nr_txn_record_log_event(txn, LOG_EVENT_PARAMS, &appv);
+  tlib_pass_if_int_equal("happy path, event seen", 1,
+                         nr_log_events_number_seen(txn->log_events));
+  tlib_pass_if_int_equal("happy path, event saved", 1,
+                         nr_log_events_number_saved(txn->log_events));
+  log_event_json = nr_log_events_get_event_json(txn->log_events, 0);
+  tlib_pass_if_not_null("happy path, event recorded", log_event_json);
+  expected = "[{"
+              "\"message\":\"" LOG_MESSAGE "\","
+              "\"log.level\":\"" LOG_LEVEL "\","
+              "\"timestamp\":" NR_STR2(LOG_TIMESTAMP)","
+              "\"trace.id\":\"0000000000000000\","
+              "\"span.id\":\"0000000000000000\","
+              "\"entity.guid\":\"" APP_ENTITY_GUID "\","
+              "\"entity.name\":\"" APP_ENTITY_NAME "\","
+              "\"hostname\":\"" APP_HOST_NAME "\""
+          "}]";
+  tlib_pass_if_str_equal("happy path, event recorded, json ok", expected,
+                         log_event_json);
+  test_txn_metric_is("happy path, event recorded, metric created",
+                     txn->unscoped_metrics, MET_FORCED, "Logging/lines", 1, 0,
+                     0, 0, 0, 0);
+  test_txn_metric_is("happy path, event recorded, metric created",
+                     txn->unscoped_metrics, MET_FORCED,
+                     "Logging/lines/" LOG_LEVEL, 1, 0, 0, 0, 0, 0);
+  tlib_pass_if_null(
+      "Logging/Forwarding/Dropped not created",
+      nrm_find(txn->unscoped_metrics, "Logging/Forwarding/Dropped"));
+  nr_txn_destroy(&txn);
+
+  /* Happy path with sampling */
```

Comment:
Could you add a case with a pool size of 0 for log events and test sampling/events not being created?

Reply:
@mfulb Addressed in 486ce76a699939e65c40d50fdcc3d9affa6063ec

Reply:
Addressed in 486ce76a699939e65c40d50fdcc3d9affa6063ec